epictrack-web/src/components/shared/__test__/Restricted.cy.tsx

```diff
@@ -0,0 +1,59 @@
+import { Provider } from "react-redux";
+import { store } from "../../../store";
+import { Restricted } from "../restricted";
+import { ROLES } from "../../../constants/application-constant";
+import { userDetails } from "services/userService/userSlice";
+
+declare global {
+  interface Window {
+    store: any; // replace 'any' with the type of your Redux store if you have one
+  }
+}
+
+describe("Restricted Component", () => {
+  beforeEach(() => {
+    if (window.Cypress) {
+      // expose the store to the Cypress window object when running Cypress tests
+      window.store = store;
+    }
+
+    const fakeUserDetail = {
+      sub: "fakeSub",
+      groups: ["fakeGroup1", "fakeGroup2"],
+      preferred_username: "fakeUsername",
+      firstName: "fakeFirstName",
+      lastName: "fakeLastName",
+      email: "fakeEmail@example.com",
+      staffId: 123,
+      phone: "1234567890",
+      position: "fakePosition",
+      roles: [ROLES.CREATE, "fakeRole1", "fakeRole2"],
+    };
+
+    cy.window().its("store").invoke("dispatch", userDetails(fakeUserDetail));
```

Comment:
wow, good job. I was wondering how we will have to test with redux. Beautiful solution

Reply:
Thanks :)